## revenue from app ads.
### how to revenue from app ads:
   - our aim to help developers to understand the behavior of users in `google lay` & `play store`. And why they can play on apps a large time without exiting the game. And how to make them play more to achieve more money by more ads.
   
  ** So we'll need to collect and analyze data about mobile apps available on Google Play and the App Store**.

In [121]:
###the apple store dataset ###

open_file=open('AppleStore.csv')
from csv import reader
read_file= reader(open_file)
AppleStore_data=list(read_file)
AppleStore_header=AppleStore_data[0]
AppleStore=AppleStore_data[1:]

* we open the Apple Store [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps):

    -open the file data and assign the data to a variable called 'open_file'
    -recall csv to import reader
    -then read the data and assign it to a variable called 'read_file'
    -assign the data as a list of the lists in a variable called 'AppledStore_data'
    -assign the header to a variable 'AppleStore_header'
    -assign all data without the header to a variable called 'AppleStore'.
    
    you can download the data from [here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)

In [122]:
###the google play dataset ###

open_file=open('googleplaystore.csv')
from csv import reader
read_file= reader(open_file)
googleplay_data=list(read_file)
googleplay_header=googleplay_data[0]
googleplay=googleplay_data[1:]

* we open the google play [A data set](https://www.kaggle.com/lava18/google-play-store-apps) :

    -open the file data and assign the data to a variable called 'open_file'
    -recall csv to import reader
    -then read the data and assign it to a variable called 'read_file'
    -assign the data as a list of the lists in a variable called 'googleplay_data'
    -assign the header to a variable 'googleplay_header'
    -assign all data without the header to a variable called 'googleplay'.
    
    you can download the data from [here](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).

* we make a function that extract the data which you want.
    -Just put your  index number which you want to start from it, 
    -and the index number of of the end data you want.

In [123]:
def explore_data(dataset, start, end, row_and_column= False):
    file_data= dataset[start : end]
    for row in file_data:
        print(row)
        print('\n')
        
    if row_and_column:
        print('the number of rows:',len(dataset))
        print('the number of column:',len(dataset[0]) )

* we define a new function call "explore_data", inside the fuction we make additional option if you want the number of'column'& 'rows'.

    -assign the data you want to a variable call 'file_data'.
    
    -use loop in side the 'file_data'
    -inside the loop print the 'row', which mean the data you want.
         -then insert a new line for each iteration.
    -use if condtion outside for loop: **it use when row_and_column=True**
       -print the number of rows
       -print the number of column
        

In [124]:
print(AppleStore_header)
print('\n')
explore_data(AppleStore,0,3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


the number of rows: 7197
the number of column: 16


* the apple store has "7197 app" & '16 column'. The column maybe useful is 'track_name' & 'size_bytes' & 'currency' & 'price' &  'prime_genre' & 'rating_count_tot' and 'rating_count_ver'.


### To understand what is mean by each column:

| Column             | Description                                     |
| ---------------    | ---------------------------------------------   |
| "id"               | App ID                                          |
| "track_name"       | App Name                                        |
| "size_bytes"       | size(in bytes)                                  |
| "currency"         | Currency Type                                   |
| "price"            | price amount                                    |
| "rating_count_tot" | User Rating counts (for all version)            |
| "rating_count_ver" | User Rating counts (for current version)        |
| "user_rating"      | Average User Rating value (for all version      |
| "user_rating_ver"  | Average User Rating value (for current version) |
| "ver"              | Latest version code                             |
| "cont_rating"      | Content Rating                                  |
| "prime_genre"      | Primary Genre                                   |
| "sup_devices.num"  | Number of supporting devices                    |
| "ipadSc_urls.num"  | Number of screenshots showed for display        |
| "lang.num"         | Number of supported languages                   |
| "vpp_lic"          | Vpp Device Based Licensing Enabled              |


In [125]:
print(googleplay_header)
print('\n')
explore_data(googleplay,0,2,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


the number of rows: 10841
the number of column: 13


* The google play has "10841 app".And the columns maybe useful is 'App' & 'Category' & 'Rating' & 'Reviews' & 'Installs' & 'Type' & 'Price' & 'Genres'.


## To find out if we have an error "missing column" in the dataset :

In [126]:
def missing_column(data_set):
    incorrect=0
    correct=0
    for row in data_set[1:]:
        if len(row) != len(data_set[0]):
            print(row)
            print(data_set.index(row))
            print('\n')
            incorrect+=1
            
        else:
            correct+=1
            
            
    if correct ==len(data_set[1:]):
             print('No error')  
            
    print('The number of column incorrect:',incorrect, 'From:', len(data_set[1:]))
    print('The number of column correct:',correct,'From:', len(data_set[1:]))

In [127]:
missing_column(AppleStore_data)

No error
The number of column incorrect: 0 From: 7197
The number of column correct: 7197 From: 7197


* to check AppleStore from any missing column

In [128]:
missing_column(googleplay_data)

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10473


The number of column incorrect: 1 From: 10841
The number of column correct: 10840 From: 10841


* to check googleplay from any missing column

# Delet Wrong Data:

the google play data set has a dedeicatd [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion). and you can see that [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015).
you will see wrong data at 10472 has an error '4.0M' can't convert to float. So, we delet this row.

# To find if we have an error "duplicate entries" in dataset:

## part 1:

In [129]:
for row in googleplay_data[1:]:
    name=row[0]
    if name == 'Instagram':
        print(row)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


## Part2

In [130]:
def deplicate_e(data_entries):
    deplicate=[]
    unique=[]
    for app in data_entries[1:]:
        name = app[0]
        if name in unique:
            deplicate.append(name)
        else:
            unique.append(name)
            
    print('the num of deplicated app:', len(deplicate)) 
    print('\n')
    print('the example of deplicate app:','\n',deplicate[:15])
    print('\n')
    print('the num of unique app:', len(unique))
    
    if deplicate == 0:
        print('No deplicated error')

In [131]:
deplicate_e(AppleStore_data)

the num of deplicated app: 0


the example of deplicate app: 
 []


the num of unique app: 7197


### after check the deplicated app in AppleStore:
    -there is no deplicated app
    -there are 7197 unique app

In [132]:
deplicate_e(googleplay_data)

the num of deplicated app: 1181


the example of deplicate app: 
 ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


the num of unique app: 9660


### after check the deplicated app in googleplay:
    - there are 1181 apps delicated
    - there are 9659 apps unique
    
** I won't remove the deplicate data randomly, i will use the criterion to do that. the criterion will be "the higher number of reviews,the more recent the data should be".**    

In [133]:
 del googleplay_data[10473]

In [134]:
reviews_max={} #it has an unique app name, the value is the highest number of reviews 

for row in googleplay_data[1:]:
    name= row[0]
    n_reviews=float(row[3])
    
    if name not in reviews_max:
        reviews_max[name]= n_reviews
    
    elif name in reviews_max and reviews_max[name]< n_reviews:
        reviews_max[name]=n_reviews
        
    
len(reviews_max)    

9659

# we try to remove the deplicated data by using a dictionary:

##  Part 1
    -first we defined a dicionary.
    -using loop through googlre play data set
    -assign the name of apps to a variable called 'name'
    -assign the number of reviews to a veriable called 'n_reviews'
    -use **if** condition to assign the name of apps to the dictionary at the first time.
    -use **elif** condition to assign the high value of the reviews in the value of dictionary if the name of the app deplicated but has a different review values.

To check if there is any error in the dataset in row[3] if it can not convert to float

In [135]:
for row in googleplay_data[1:]:

    if n_reviews != float(row[3]):
        print(googleplay.index(row))
    
    

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
222
230
231
232
233
234
235
204
237
238
193
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
213
257
258
259
260
253
262
263
264
204
237
238
193
240
241
242
243
244
245
246
247
24

In [136]:
android_clean=[] #it will contain the data cleaned and the other data related to the same row
already_added=[] #it will contain the apps name 
for row in googleplay_data[1:]:
    name=row[0]
    n_reviews=float(row[3])
    if n_reviews == reviews_max[name] and (name not in already_added):
        already_added.append(name)
        android_clean.append(row)        

# let's see the new data:
- after removing the deplicated data in google play data set.
- you can see the difference in the number of rows. **it decreased from 10841 to 9659**


In [137]:
explore_data(android_clean,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


the number of rows: 9659
the number of column: 13


## Part 2:
    -we define an empty list 'android_clean' which will contain the data cleaned antd the remain data which related to the app.
    -we define an empty list 'alredy_added' which will has the unique name of app only without any deplicated.
    -use loop in google play data set.
    -assign the name of apps tho a variable called 'name'.
    -assign the number of reatings to a variable called 'n_reviews'.
    -use **if** condtion to assign app name to a list called 'already_added' if not exsit in it. and the name has the same review value. by using append().
    -assing all clean data to a list 'android_clean' by append().

# for remove non-english app:
    -we try to show some examples about non-english apps in the "part1", which doesn't need to use in my cleaning data.
    -in the "part2" we will make a function that can show us the app name is in english language or in another language.By typing "True" or "False"

## part1: example for non-english app

In [138]:
print(AppleStore_data[814][1])
print(AppleStore_data[6732][1])
print('\n')
print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


中国語 AQリスニング
لعبة تقدر تربح DZ


## part 2: we wrote a function that detects non-English app names

### in this code:

-we make a new function that takes in the string and returns "True if the string in english language  or has at maximum 3 (emojis or other symbols) " or "False if the string in another langauge or has more than 3 (emojis or other symbols) ".
    -we use loop in the string or "app name".
    
     - then use if to check the ord() of the string is more than 127 or less than it.
    -if the ord() of the string is higher than 127 it return "False". and if the ord() of the string is lower than or equal 127 it return "True"

In [139]:
def App_english(app_name):
    non_english=0
    for char in app_name:
        if ord(char)  > 127 :
            non_english+=1
            
    if non_english > 3:
        return False
    else:
         return True 

* this function is not perfect, so we will use it as a part of solution. to find the perfect a filter

In [140]:
print(App_english('Instagram'))
print(App_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(App_english('Docs To Go™ Free Office Suite'))
print(App_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

True
False
True
True
8482
128540


* in the first code the string is fully english so it shows as "True"
* in the second code the string has a Strange symbol so it shows as "False"
* in the third code the ord() of " ™ "=8482 which higher than 127 so it shows as "False".
* ih the fourth code the ord() of " 😜"= 128540 which higher than 127 so it shows as "False".

# we will use the new function to filter out non-english apps

In [141]:
android_english=[]
IOS_english=[]

for row in android_clean:
    name= row[0]     
    if  App_english(name):
        android_english.append(row) 

for row in AppleStore_data[1:]:
    name= row[1]     
    if  App_english(name):
        IOS_english.append(row)
    

In [142]:
explore_data(android_english,0,3,True)
print('\n')
explore_data(IOS_english,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


the number of rows: 9614
the number of column: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12',

In [143]:
free_android=[]
free_IOS=[]

for row in android_english:
    price=row[7]
    if price =='0':
        free_android.append(row)
        
for row in IOS_english:
    price=row[4]
    if price == '0.0':
        free_IOS.append(row)        

In [144]:
explore_data(free_android,0,3,True)
print('\n')
explore_data(free_IOS,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


the number of rows: 8864
the number of column: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12',

* we want to find an app profile that fits to 'google play' & 'Apple Store'.
Because the final goal to add the app to google paly and Apple Store. So, we need to find a successful app on both market. We will use a strategy to find it.
### our strategy is :
    1. to build the minimal value of app version, and add it to 'google play'
    2. if the app has a good response from users, we will develop it further.
    3. If the app come a profitable after six months, we will build ios version and and it to 'Apple Store'.

# why we want to find a profile app that fits to the both market:
     -Because, our revenue is highly influenced by the number of people using our apps. To minimize risks and overhead, we follow a strategy comprised of three steps:
     
     1. Build a minimal Android version of the app, and add it to 'Google Play' 
     2. If the app has a good response from users, we develop it further.
     3. If the app is profitable after six months, we build iOS version and add it to 'Apple Store'.
 
## we do that by two steps:
    1. cleaning data
    2. analysis data

### now, we will start in a sept two 'Analysis data'

In [145]:
#free_android=[]
#free_IOS=[]

def freq_table(data_clean, index):
    table={}
    total=0
    
    for row in data_clean:
        total+=1
        value = row[index] 
        if value in table:
            table[value]+=1
        else:
            table[value]=1
            
            
    table_precentages={}
    
    for key in table:
        precentage=(table[key]/total)*100
        table_precentages[key]=precentage
    return table_precentages
            
            
        
def display_table(data_clean, index):
    table = freq_table(data_clean, index)
    display = []
    for key in table:
        key_value=(table[key],key)
        display.append(key_value)
        
    table_stored= sorted(display,reverse = True)
    for row in table_stored:
        print(row[1], ':', row[0])
        

In [146]:
display_table(free_IOS , -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


(as we can see in the free english app) the percentage of Games is 58.16% it's the most common 'Apple Store', then the percentage of Entertainment is 7.88% almost 8%, then the percentage of Photo & Vedio is almost 5%, then the percentage of Eduction is almost 4% and then the Social Networking is 3.29%.

So, The general impression in App Store is  the most common apps is for fun like(games, entertainment, photo & vedio,..)And the techincal fields with rare percentages like(sports , music , education,..)

-But this percentage not represent the number of users.

In [147]:
display_table(free_android , 1 ) # Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

On the opposite way in Google Play: the most common category in Google Play is 1st Family(like games for kids) by almost 19%, then 2end Games by 9.72% , then 3th Tools by 8.46%, then 4th Business by 4.5%, then 5th Life style by 3.9%.

So, the most common category for games & practical purposes.

The pratical apps seems have to better representation on Google Play compared to App Store.

In [148]:
display_table(free_android , -4 ) #Genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

### Recommed:
Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

In [149]:
IOS_genre= freq_table(free_IOS, -5)

for genre in IOS_genre:
    total=0
    len_genre=0
    
    for app in free_IOS:
        genre_app=app[-5]
        
        if genre_app == genre:
            n_rating=float(app[5])
            total+=n_rating
            len_genre+=1
            
    average=total/len_genre
    print(genre,':',average)

Shopping : 26919.690476190477
Travel : 28243.8
Sports : 23008.898550724636
Music : 57326.530303030304
News : 21248.023255813954
Weather : 52279.892857142855
Reference : 74942.11111111111
Medical : 612.0
Business : 7491.117647058823
Catalogs : 4004.0
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Food & Drink : 33333.92307692308
Education : 7003.983050847458
Games : 22788.6696905016
Productivity : 21028.410714285714
Entertainment : 14029.830708661417
Finance : 31467.944444444445
Utilities : 18684.456790123455
Photo & Video : 28441.54375
Social Networking : 71548.34905660378
Book : 39758.5
Health & Fitness : 23298.015384615384


as we see the Navigation has the higher average number of reviews, but i recommende the Soial Networking & Games:

In [150]:
for app in free_IOS:
    if app[-5] =='Social Networking': #genre_app=app[-5]
        print(app[1],':',app[5]) #app_name=app[1], n_reviews=app[5]

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

the Social Network may be the 3th in average number but but this figure is heavily influenced by facebook , Pinterest. it increased more than 4 million reviews.


In [168]:
for app in free_IOS:
    if app[-5] =='Games': #genre_app=app[-5]
        print(app[1],':',app[5]) #app_name=app[1], n_reviews=app[5]

Clash of Clans : 2130805
Temple Run : 1724546
Candy Crush Saga : 961794
Angry Birds : 824451
Subway Surfers : 706110
Solitaire : 679055
CSR Racing : 677247
Crossy Road - Endless Arcade Hopper : 669079
Injustice: Gods Among Us : 612532
Hay Day : 567344
PAC-MAN : 508808
DragonVale : 503230
Head Soccer : 481564
Despicable Me: Minion Rush : 464312
The Sims™ FreePlay : 446880
Sonic Dash : 418033
8 Ball Pool™ : 416736
Tiny Tower - Free City Building : 414803
Jetpack Joyride : 405647
Bike Race - Top Motorcycle Racing Games : 405007
Kim Kardashian: Hollywood : 397730
Trivia Crack : 393469
WordBrain : 391401
Sniper 3D Assassin: Shoot to Kill Gun Game : 386521
Flow Free : 373857
Geometry Dash Lite : 370370
▻Sudoku : 359832
Fruit Ninja® : 327025
Pixel Gun 3D : 301182
Temple Run 2 : 295211
My Horse : 293857
Word Cookies! : 287095
Dragon City Mobile : 277268
The Simpsons™: Tapped Out : 274501
Plants vs. Zombies™ 2 : 267394
Clash Royale : 266921
Pokémon GO : 257627
CSR Racing 2 : 257100
Star Wars™: 

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

## Most Popular Apps by Genre on Google Play


In [152]:
display_table(free_android , 5 ) 

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

In [153]:
android_category = freq_table(free_android,1)
for category in android_category:
    total=0
    len_category=0
    
    for app in free_android:
        category_app= app[1]
        
        if category_app == category:
            install_num = app[5]
            install_num = install_num.replace('+','')
            install_num = install_num.replace(',','')
            total+=float(install_num)
            len_category+=1
            
    average=total/len_category 
    print(category,':',average)

AUTO_AND_VEHICLES : 647317.8170731707
HOUSE_AND_HOME : 1331540.5616438356
LIFESTYLE : 1437816.2687861272
SOCIAL : 23253652.127118643
BUSINESS : 1712290.1474201474
GAME : 15588015.603248259
TOOLS : 10801391.298666667
ENTERTAINMENT : 11640705.88235294
COMICS : 817657.2727272727
BOOKS_AND_REFERENCE : 8767811.894736841
FAMILY : 3695641.8198090694
FINANCE : 1387692.475609756
PHOTOGRAPHY : 17840110.40229885
MEDICAL : 120550.61980830671
ART_AND_DESIGN : 1986335.0877192982
TRAVEL_AND_LOCAL : 13984077.710144928
EDUCATION : 1833495.145631068
VIDEO_PLAYERS : 24727872.452830188
PRODUCTIVITY : 16787331.344927534
EVENTS : 253542.22222222222
COMMUNICATION : 38456119.167247385
SPORTS : 3638640.1428571427
BEAUTY : 513151.88679245283
MAPS_AND_NAVIGATION : 4056941.7741935486
HEALTH_AND_FITNESS : 4188821.9853479853
DATING : 854028.8303030303
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
NEWS_AND_MAGAZINES : 9549178.467741935
FOOD_AND_DRINK : 1924897.7363636363
PERSONALIZATION : 52014

i recommended the shopping because, the people spend alot of time to see ads and offer. From that we can get revenue.

In [165]:
for app in free_android:
    if app[1] =='GAME': #genre_app=app[-5]
        print(app[0],':',app[5]) #app_name=app[1], n_reviews=app[5]

Solitaire : 10,000,000+
Sonic Dash : 100,000,000+
PAC-MAN : 100,000,000+
Bubble Witch 3 Saga : 50,000,000+
Race the Traffic Moto : 10,000,000+
Marble - Temple Quest : 10,000,000+
Shooting King : 10,000,000+
Geometry Dash World : 10,000,000+
Jungle Marble Blast : 5,000,000+
Roll the Ball® - slide puzzle : 100,000,000+
Block Craft 3D: Building Simulator Games For Free : 50,000,000+
Farm Fruit Pop: Party Time : 1,000,000+
Love Balls : 50,000,000+
Piano Tiles 2™ : 100,000,000+
Pokémon GO : 100,000,000+
Paint Hit : 10,000,000+
Snake VS Block : 50,000,000+
Rolly Vortex : 10,000,000+
Woody Puzzle : 1,000,000+
Stack Jump : 10,000,000+
The Cube : 5,000,000+
Extreme Car Driving Simulator : 100,000,000+
Bricks n Balls : 1,000,000+
The Fish Master! : 1,000,000+
Color Road : 10,000,000+
Draw In : 10,000,000+
PLANK! : 500,000+
Looper! : 1,000,000+
Trivia Crack : 100,000,000+
Will it Crush? : 5,000,000+
Tomb of the Mask : 5,000,000+
Baseball Boy! : 10,000,000+
Hello Stars : 10,000,000+
Tank Stars : 1

the games have a large number of install so we will divide it to:
    1. up 1000,000,000+ to 100,000,000
    2. form 1000,000 to 50,000,000

## up 1000,000,000+ to 100,000,000

In [166]:
for app in free_android:
    if app[1] =='SHOPPING' and ( app[5]=='500,000,000+'
                            or app[5]=='100,000,000+'
                            or app[5]=='50,000,000+'):
    
        print(app[0],':',app[5])

eBay: Buy & Sell this Summer - Discover Deals Now! : 100,000,000+
letgo: Buy & Sell Used Stuff, Cars & Real Estate : 50,000,000+
Amazon Shopping : 100,000,000+
Lazada - Online Shopping & Deals : 50,000,000+
OLX - Buy and Sell : 50,000,000+
The birth : 50,000,000+
Mercado Libre: Find your favorite brands : 50,000,000+
Myntra Online Shopping App : 50,000,000+
Flipkart Online Shopping App : 100,000,000+
AliExpress - Smarter Shopping, Better Living : 100,000,000+
Wish - Shopping Made Fun : 100,000,000+
Groupon - Shop Deals, Discounts & Coupons : 50,000,000+



However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 40,000,000 downloads):

In [167]:
for app in free_android:
    if app[1] =='GAME' and ( app[5]=='1,000,000+'
                            or app[5]=='5,000,000+'
                            or app[5]=='10,000,000+'
                            or app[5]=='40,000,000+'):
    
        print(app[0],':',app[5])

Solitaire : 10,000,000+
Race the Traffic Moto : 10,000,000+
Marble - Temple Quest : 10,000,000+
Shooting King : 10,000,000+
Geometry Dash World : 10,000,000+
Jungle Marble Blast : 5,000,000+
Farm Fruit Pop: Party Time : 1,000,000+
Paint Hit : 10,000,000+
Rolly Vortex : 10,000,000+
Woody Puzzle : 1,000,000+
Stack Jump : 10,000,000+
The Cube : 5,000,000+
Bricks n Balls : 1,000,000+
The Fish Master! : 1,000,000+
Color Road : 10,000,000+
Draw In : 10,000,000+
Looper! : 1,000,000+
Will it Crush? : 5,000,000+
Tomb of the Mask : 5,000,000+
Baseball Boy! : 10,000,000+
Hello Stars : 10,000,000+
Tank Stars : 10,000,000+
Hole.io : 10,000,000+
Mini Golf King - Multiplayer Game : 5,000,000+
Flip the Gun - Simulator Game : 10,000,000+
Mad Skills BMX 2 : 1,000,000+
MMX Hill Dash 2 – Offroad Truck, Car & Bike Racing : 1,000,000+
Word Link : 10,000,000+
Last Day on Earth: Survival : 10,000,000+
Partymasters - Fun Idle Game : 10,000,000+
Harry Potter: Hogwarts Mystery : 10,000,000+
Offroad Outlaws : 1,0


This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.


## Conclusions
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

so, my plan is to make a simple game which can attract the kids to play a large time and we can add a meduim & hard level for other childern 